File: csharp/cars-assemble/CarsAssemble.cs
using System;

static class AssemblyLine
{
    private const int CarsPerHour = 221;

    public static double ProductionRatePerHour(int speed) =>
        SuccessRate(speed) * CarsPerHour * speed;
    
    private static double SuccessRate(int speed) =>
        speed switch
        {
            >= 1 and <= 4 => 1.0,
            >= 5 and <= 8 => 0.9,
            9 => 0.8,
            10 => 0.77,
            _ => 0.0
        };

    public static int WorkingItemsPerMinute(int speed) =>
        (int) (ProductionRatePerHour(speed) / 60);
}

File: r/cars-assemble/cars-assemble.R
success_rate <- function(speed) {
  if (speed == 0) return(0.0)
  if (speed <= 4) return(1.0)
  if (speed <= 8) return(0.9)
  if (speed == 9) return(.8)
  0.77
}

production_rate_per_hour <- function(speed) {
  221 * speed * success_rate(speed)
}

working_items_per_minute <- function(speed) {
  production_rate_per_hour(speed) %/% 60
}
